mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -915,6 +917,22 @@ class GeneralizedLinearRegressionSummary private[regression] (
   /** Number of instances in DataFrame predictions. */
   private[regression] lazy val numInstances: Long = predictions.count()
 
+
+  /**
+   * Name of features. If the name cannot be retrieved from attributes,
+   * set default names to "V1", "V2", and so on.
+   */
+  @Since("2.2.0")
+  lazy val featureName: Array[String] = {
+    val featureAttrs = AttributeGroup.fromStructField(
+      dataset.schema(model.getFeaturesCol)).attributes
+    if (featureAttrs == None) {
+      Array.tabulate[String](origModel.numFeatures)((x: Int) => ("V" + (x + 1)))
```

Comment:
is there another way feature names could be used elsewhere? the concern is names from here would not match feature names in other pleases if this is not stored in the model

might need to add more tests for this?